kubernetes/config/incluster_config.py

```diff
@@ -0,0 +1,86 @@
+# Copyright 2016 The Kubernetes Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import os
+
+from kubernetes.client import configuration
+
+_SERVICE_HOST_ENV_NAME = "KUBERNETES_SERVICE_HOST"
+_SERVICE_PORT_ENV_NAME = "KUBERNETES_SERVICE_PORT"
+_SERVICE_TOKEN_FILENAME = "/var/run/secrets/kubernetes.io/serviceaccount/token"
+_SERVICE_CERT_FILENAME = "/var/run/secrets/kubernetes.io/serviceaccount/ca.crt"
+
+
+def _join_host_port(host, port):
+    """Adapted golang's net.JoinHostPort"""
+    template = "%s:%s"
+    host_requires_bracketing = ':' in host or '%' in host
+    if host_requires_bracketing:
+        template = "[%s]:%s"
+    return template % (host, port)
+
+
+class ConfigException(Exception):
+    pass
+
+
+class InClusterConfigLoader(object):
+    def __init__(self, host_env_name, port_env_name, token_filename,
+                 cert_filename, environ=os.environ):
+        self._host_env_name = host_env_name
+        self._port_env_name = port_env_name
+        self._token_filename = token_filename
+        self._cert_filename = cert_filename
+        self._environ = environ
+
+    def load_and_set(self):
+        self._load_config()
+        self._set_config()
+
+    def _load_config(self):
+        if (self._host_env_name not in self._environ or
```

Comment:
What about the case where either of the env vars are defined but empty (e.g. ``KUBERNETES_SERVICE_HOST=``)?